tests/common.py

```diff
@@ -542,27 +542,36 @@ def __init__(self, root, *args):
         self.root = root
         self.submodules = args
 
-    def __call__(self, func):
-        """Apply decorator."""
-        from unittest.mock import MagicMock, patch
-
+    def __enter__(self):
```

Comment:
Rewriting `MockDependency` so that it can be used as both a decorator and a context manager.